### PR TITLE
Test/local mappers

### PR DESCRIPTION
**Pull request by saraashraf11**
# Pull Request Template

## Description

This Pull Request adds unit test coverage for the following components in the repository layer. The purpose of this PR is to ensure full test coverage and validate the mapping logic between local Room entities and domain entities.

---

## Changes Made

- Implemented unit tests for TvShowWithCategoryLocalMapper
- Created fake data classes and helpers for testing
-  Covered edge cases and ensured correctness of mapped data
- Completed test coverage for 10 test files related to local mappers

---

## Screenshots 

image

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

### CODE REVIEW

**Review by marah222 — DISMISSED** [review]

File: repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ private lateinit var mapper: MovieGenreLocalMapper
+
+ @BeforeEach
+ fun setUp() {
+  mapper = MovieGenreLocalMapper()
+ }
+
+ @Test
+ @DisplayName("should return MovieGenre.ALL when categoryId is out of range")
+ fun `toEntity should return ALL genre when categoryId is invalid`() {
```
Comment (Noor-Serry) [inline repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt]: you have only one function in the class so no need to specifiy the function name also no need for display name annotation

Comment (ThoriaHamdy99) [reply repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt]: I think even if there is one function you need to specify what this function do by its name.
So if you update the implementation you needto know if this function will be enough or you need another function to handle your changes

File: repository/src/test/java/com/example/repository/mapper/local/MovieLocalMapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ @BeforeEach
+ fun setUp() {
+  mapper = MovieLocalMapper()
+ }
+
+ @Test
+ @DisplayName("should return Movie entity when converting from LocalMovieDto")
+ fun `toEntity should return Movie when given LocalMovieDto`() {
+  // Arrange
+  val dto = LocalMovieDto(
```
Comment (ThoriaHamdy99) [inline repository/src/test/java/com/example/repository/mapper/local/MovieLocalMapperTest.kt]: Please you can make this decleration outside in seperate file make a create function with default values and pass the needed values to it.

this change for all object declerations here

File: repository/src/test/java/com/example/repository/mapper/local/MovieLocalMapperTest.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+   originCountry = "USA",
+   hasVideo = true
+  )
+
+  // Act
+  val result = mapper.toEntity(dto)
+
+  // Assert
+  assertThat(result.id).isEqualTo(101)
+  assertThat(result.name).isEqualTo("Inception")
```
Comment (ThoriaHamdy99) [inline repository/src/test/java/com/example/repository/mapper/local/MovieLocalMapperTest.kt]: you can make one assertion instead of all of this.
this comment also for all your test functions

File: repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt (hunk 1 above)
Comment (eslam-99) [reply repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt]: I agree with Thoraya

**Review by ThoriaHamdy99 — CHANGES_REQUESTED** [review]
I have just 2 comments:
- You can make the variables decleration outside the test class in seperate file make a create function with default values and pass the needed values to it.
- You can make one assertion only in each test function.

File: repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt (hunk 1 above)
Comment (Noor-Serry) [reply repository/src/test/java/com/example/repository/mapper/local/MovieGenreLocalMapperTest.kt]: if I change the class then I should change the test .
you can check this repo by philipplackner :
https://github.com/philipplackner/CleanArchitectureNoteApp/blob/testing/app/src/test/java/com/plcoding/cleanarchitecturenoteapp/feature_note/domain/use_case/GetNotesTest.kt

**Review by ThoriaHamdy99 — DISMISSED** [review]

**Review by eslam-99 — DISMISSED** [review]

**Review by ThoriaHamdy99 — APPROVED** [review]

**Review by eslam-99 — APPROVED** [review]

**Review by Ahmed-AlMancy — APPROVED** [review]
